sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/query.sql

```diff
@@ -0,0 +1,367 @@
+),
+stub_other_dl AS (
+  SELECT
+    stub_visit_id,
+    stub_download_session_id,
+    CASE
+      WHEN (COUNT(*) > 1)
+        THEN TRUE
+      ELSE FALSE
+    END AS additional_download_occurred
+  FROM
+    stub_dl
+  GROUP BY
+    stub_visit_id,
+    stub_download_session_id
+),
+stub AS (
+  SELECT
+    stub_visit_id,
+    stub_download_session_id,
+    dltoken,
+    count_dltoken_duplicates,
+    additional_download_occurred,
+    download_date
+  FROM
+    stub_dl
+  JOIN
+    stub_other_dl
+  USING
+    (stub_visit_id, stub_download_session_id)
+),
+-- This will drop all the ga_sessions w/o a DLtoken but keep DLtoken without a GA session.
+-- This will also result in multiple rows as the ga.client_id is not unique for the day
+-- since this visit_id is missing from the stub.
+-- The join must use client_id/stub_visit_id and download_session_id/stub_download_session_id
+downloads_and_ga_session AS (
+  SELECT
+    gs.client_id AS client_id,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(country)) AS country,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(ad_content)) AS ad_content,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(campaign)) AS campaign,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(medium)) AS medium,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(source)) AS source,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(device_category)) AS device_category,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(os)) AS os,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(browser)) AS browser,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(browser_version)) AS browser_version,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(`language`)) AS `language`,
+    s.stub_visit_id AS stub_visit_id,
+    s.stub_download_session_id AS stub_download_session_id,
+    s.dltoken AS dltoken,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(landing_page)) AS landing_page,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(pageviews)) AS pageviews,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(unique_pageviews)) AS unique_pageviews,
+    LOGICAL_OR(
+      has_ga_download_event
+    ) AS has_ga_download_event,  -- this will be ignored if nrows >1
+    mozfun.stats.mode_last_retain_nulls(
+      ARRAY_AGG(count_dltoken_duplicates)
+    ) AS count_dltoken_duplicates,
+    LOGICAL_OR(additional_download_occurred) AS additional_download_occurred,
+    COUNT(*) AS nrows,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(s.download_date)) AS download_date,
+    mozfun.stats.mode_last_retain_nulls(ARRAY_AGG(time_on_site)) AS time_on_site
+  FROM
+    ga_sessions_with_hits gs
+  RIGHT JOIN
+    stub s
+  ON
+    (gs.client_id = s.stub_visit_id AND gs.download_session_id = s.stub_download_session_id)
+  GROUP BY
+    gs.client_id,
+    dltoken,
+    stub_visit_id,
+    stub_download_session_id
+)
+SELECT
+  dltoken,
+  download_date,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    ELSE time_on_site
+  END
```

Comment:
nit: This could be simpler with e.g. `IF(nrows <= 1, itme_on_site, NULL)`

Reply:
Yep - will update

Reply:
PR updated.